File: projectEuler/13/13.py
#Work out the first ten digits of the sum of the following one-hundred 50 digit numbers.

#add your path to the file
path = 'projectEuler/13/13.txt'
with open(path, 'r') as f:
    #read the numbers line by line
    numbers = f.readlines()

def sum_of_numbers_digits(numbers):
    sum = 0
    for number in numbers:
        sum += int(number)
    return sum

print(str(sum_of_numbers_digits(numbers))[:10])

File: projectEuler/13/13.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

// Function definition
std::string sumOfNumbersDigits(const std::vector<std::string>& numbers) {
    unsigned long long totalSum = 0;

    for (std::vector<std::string>::const_iterator it = numbers.begin(); it != numbers.end(); ++it) {
        totalSum += std::stoull(it->substr(0, 15)); // Take the first 15 digits of the number
    }

    return std::to_string(totalSum).substr(0, 10); // Return the first 10 digits of the total sum
}

int main() {
    // Add your txt file path here
    std::string path = "projectEuler/13/13.txt"; // Path variable for the file
    std::ifstream file(path); // Use the path variable to open the file
    std::vector<std::string> numbers;
    // Read the file line by line and store the numbers in a vector
    std::string line;

    if (file.is_open()) {
        while (std::getline(file, line)) {
            numbers.push_back(line);
        }
        file.close();
    } else {
        std::cerr << "File could not be opened: " << path << std::endl;
        return 1;
    }

    std::cout << sumOfNumbersDigits(numbers) << std::endl;

    return 0;
}